var/spack/repos/builtin/packages/libvori/package.py

```diff
@@ -19,3 +19,11 @@ class Libvori(CMakePackage):
     version("201229", sha256="da0afb292c94f8de2aaebfd0b692d15ffd86083cb8a48478b07ca93823decc06")
     version("201224", sha256="16f6c49eaa17ea23868925dbaae2eca71bdacbe50418c97d6c55e05728038f31")
     version("201217", sha256="6ad456ed6ca5d28cadcc0d90eabe8fff5caa77b99f12764323de5e3ae21cddf5")
+
+    variant("fpic", default=True, description="Compile the library with position independent code")
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        args = [self.define_from_variant("CMAKE_POSITION_INDEPENDENT_CODE", "fpic")]
```

Comment:
There are spack keys like “self.compiler.cxx_pic_flag”, "self.compiler.cc_pic_flag" or "self.compiler.f77_pic_flag" to set PIC compiler flag; see superlu or other packages provide the pic variant.

Reply:
I am not setting the compiler flag explicitly. CMake does it for me as it should be. If I am not mistaken `self.compiler:.....` is used when we need to generate makefiles by hand which is not our case here.